modules/kubernetes/src/main/java/org/apache/ignite/spi/discovery/tcp/ipfinder/kubernetes/ThinClientKubernetesAddressFinder.java

```diff
@@ -0,0 +1,100 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.ignite.spi.discovery.tcp.ipfinder.kubernetes;
+
+import java.net.InetAddress;
+import java.util.function.Supplier;
+import org.apache.ignite.kubernetes.KubernetesConnectorConfigurator;
+import org.apache.ignite.kubernetes.KubernetesConnectorDefaults;
+import org.apache.ignite.kubernetes.KubernetesServiceAddressResolver;
+
+/**
+ * Address finder for automatic lookup of Ignite nodes running in Kubernetes environment. All Ignite nodes have to
+ * deployed as Kubernetes pods in order to be found. Applications and Ignite nodes running outside of Kubernetes
+ * will not be able to reach the containerized counterparts.
+ * <p>
+ * The implementation is based on a distinct Kubernetes service that has to be created and should be deployed prior
+ * Ignite nodes startup. The service will maintain a list of all endpoints (internal IP addresses) of all containerized
+ * Ignite pods running so far. The name of the service must be equal to {@link #setServiceName(String)} which is
+ * `ignite` by default.
+ * <p>
+ * As for Ignite pods, it's recommended to label them in such a way that the service will use the label in its selector
```

Comment:
This paragraph is also confusing. We don't care how users define their services and use their selectors, why mention that?

Reply:
Actually, we should care. K8S services are bounded with k8s deployments by selector. IpFinder / AddressFinder return addresses to all pods that are matches service. So, for IpFinder service should contain all server and thick clients nodes. For AddressFinder service should contain only server nodes. Otherwise there could be issues with failed connections to other pods.

Reply:
What you describe is correct, but also obvious - you have to use labels to define a service with a subset of pods in k8s. This is common knowledge and this is the first thing k8s docs tell you. So:
1. This part is redundant
2. This part is worded in such a way that (in my opinion) can bring unnecessary confusion.

Anyway, this is a minor thing - keep it if you insist.

Reply:
I think you're correct. I reword it, just add mention about nodes types that are target by service. I think this mention is important as invalid configuration would lead to performance issues.